### PR TITLE
Remove unused fields & other stuff

### PR DESCRIPTION
**Pull request by davkean**
New build of VS has more analysis, clean up the error list to remove unused fields/methods, and apply var settings.

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/ProjectSystemOptions.cs [hunk 1]
```diff
@@ -13,11 +12,8 @@
 namespace Microsoft.VisualStudio.ProjectSystem.VS
 {
     [Export(typeof(IProjectSystemOptions))]
-    internal class ProjectSystemOptions : IProjectSystemOptions
+    internal partial class ProjectSystemOptions : IProjectSystemOptions
```
Comment (drewnoakes) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/ProjectSystemOptions.cs]: Why was this made `partial`?

Comment (davkean) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/ProjectSystemOptions.cs]: Ah good catch. I rejected/modified a Roslyn code fix and it was left around from that. Fixing...

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs [hunk 2]
```diff
@@ -81,7 +78,7 @@ internal partial class LanguageServiceHost : OnceInitializedOnceDisposedAsync, I
 
         [ProjectAutoLoad(completeBy: ProjectLoadCheckpoint.ProjectFactoryCompleted)]
         [AppliesTo(ProjectCapability.DotNetLanguageService)]
-        private Task OnProjectFactoryCompletedAsync()
+        public Task OnProjectFactoryCompletedAsync()
```
Comment (jmarolf) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs]: Why is this public now?

Comment (davkean) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs]: The analyzer doesn't understand MEF exports - public makes it quiet.

**Comment by davkean** [comment]
@dotnet-bot test this please

**Comment by drewnoakes** [comment]
Merged for @davkean after GitHub's outage.

**Pull request merged**